File: C++/148. Sort List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* sortList(ListNode* head) {
        ListNode* temp=head;
        if(head==NULL){
            return NULL;
        }
        vector<int>vec;
        while(temp!=NULL){
            vec.push_back(temp->val);
            temp=temp->next;
        }
        sort(vec.begin(),vec.end());
        ListNode* res= new ListNode(vec[0]);
        ListNode* t=res;
        for(int i=1;i<vec.size();i++){
            ListNode* curr= new ListNode(vec[i]); 
            t->next=curr;
            t=curr;
        }
        t->next=NULL;
        return res;
    }
};

File: C#/148. Sort List.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode SortList(ListNode head) {
        ListNode temp=head;
        if(head==null){
            return null;
        }
        List<int> list = new List<int>();
        while(temp!=null){
            list.Add(temp.val);
            temp=temp.next;
        }
        list.Sort();
        ListNode res= new ListNode(list[0]);
        ListNode t=res;
        for(int i=1;i<list.Count;i++){
            ListNode curr= new ListNode(list[i]); 
            t.next=curr;
            t=curr;
        }
        t.next=null;
        return res;
    }
}
